cmd/plugins/balloons/policy/balloons-policy.go

```diff
@@ -1221,6 +1234,12 @@ func (p *balloons) fillFarFromDevices(blnDefs []*BalloonDef) {
 		avoidDevs = append(avoidDevs, virtDevIsolatedCpus)
 	}
 	for _, blnDef := range blnDefs {
+		if blnDef.CoreType == "efficient" {
```

Comment:
fillCloseToDevices() has already appended virtDevECores and virtDevPCores into blndef.PreferCloseToDevices. The next loop after these `if`s goes through blndef.PreferCloseToDevices and adds all devices in there into the avoidDevs list (but at most once).

This change adds virtDevECores and virtDevPCores potentially many times into the avoidDevs list, and they will be added once more by the next loop.

This sounds like unwanted behavior. Did you have some specific case (a preference order?) in mind when adding these separately, or would it suffice to let the existing loop add them. In the latter case this change can be dropped.

Reply:
Thanks, fixed.